roles/deprovision-keycloak-apb/tasks/main.yml

```diff
@@ -1,49 +1,11 @@
-- name: delete route
-  openshift_v1_route:
-    name: '{{ keycloak_route_name }}'
-    namespace: '{{ namespace }}'
-    state: absent
-
-- name: delete services
-  k8s_v1_service:
-    name: '{{ item }}'
-    namespace: '{{ namespace }}'
-    state: absent
-  with_items:
-  - '{{ keycloak_service_name }}'
-  - postgres
-
-- name: delete deployment configs
-  openshift_v1_deployment_config:
-    name: '{{ item }}'
-    namespace: '{{ namespace }}'
-    state: absent
-  with_items:
-  - keycloak
-  - postgres
-
-- name: delete secrets
-  k8s_v1_secret:
-    name: '{{ item }}'
-    namespace: '{{ namespace }}'
-    state: absent
-  with_items:
-  - keycloak
-  - keycloak-postgres
-
-- name: delete config maps
-  k8s_v1_config_map:
-    name: '{{ item }}'
-    namespace: '{{ namespace }}'
-    state: absent
-  with_items:
-  - keycloak
-
-- name: delete persistent volume claims
-  k8s_v1_persistent_volume_claim:
-    name: '{{ item }}'
-    namespace: '{{ namespace }}'
-    state: absent
-  with_items:
-  - keycloak-metrics
-  - postgres
+---
+- block:
+  - include_tasks: deprovision-keycloak.yml
+    when: USE_SHARED_SERVICE is undefined or
```

Comment:
Where is USE_SHARED_SERVICE coming from?

Reply:
It passes it as extra-vars to deprovision POD
[[ deprovision --extra-vars {"ADMIN_NAME":**********,"ADMIN_PASSWORD":**********,"SHARED_HOST":"https://keycloak-kcschared.192.168.37.1.nip.io","SHARED_REALM":"KcMyRealm","USE_SHARED_SERVICE":true,"_apb_plan_id":"default","_apb_service_class_id":"992fc0b7175fa69e2f7f12a237c78761","_apb_service_instance_id":"540bd506-225e-4eac-b0f6-ef2125dc69bb","cluster":"openshift","namespace":"kcmy "} == *\s\2\i\/\a\s\s\e\m\b\l\e* ]]

Reply:
ahh ok so it must be set in the credentials we encode during the provision. That makes sense.

Reply:
I think the order this should go in is
- clean up resources
- clean up keycloak
- deprovision

The reason for this is if someone deprovisions keycloak in the namespace where keycloak is running it will remove the running pods. This will cause an error when someone deletes the service instance from a different shared namespace. 
I think we should always clean up the resources in the current namespace and then attempt to clean up the keycloak pieces. ( basically if USE_SHARED_SERVICE is defined then handle keycloak not being there

Reply:
@maleck13 
- **clean up resources** should run always even if some of tasks fail and thats related mainly to shared and non shared keycloak instances
- **deprovision** also cleans secrets but related only to keycloak-postgres, that shoudl not delete keycloak secret. Only cleanup resources removes keycloak secret and keycloak configmap